core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -258,7 +258,10 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     // Make fake resource offers on all executors
     private def makeOffers() {
       // Make sure no executor is killed while some task is launching on it
-      val taskDescs = CoarseGrainedSchedulerBackend.this.synchronized {
+      // SPARK-27112: We need to ensure that there is ordering of lock acquisition
+      // between TaskSchedulerImpl and CoarseGrainedSchedulerBackend objects in order to fix
+      // the deadlock issue exposed in SPARK-27112
+      val taskDescs = withLock({
```

Comment:
No need for the parentheses.

Reply:
Done

Reply:
Sorry for the extra commits, was fixing code indentation.